packages/langium/src/service-registry.ts

```diff
@@ -41,44 +42,55 @@ export interface ServiceRegistry {
 export class DefaultServiceRegistry implements ServiceRegistry {
 
     protected singleton?: LangiumCoreServices;
-    protected map?: Record<string, LangiumCoreServices>;
+    protected idMap = new Map<string, LangiumCoreServices>();
+    protected extMap = new Map<string, LangiumCoreServices>();
```

Comment:
It's unfortunate that the previous `map` was protected. I'd apply this change in the 3.x version stream anyway because it brings important fixes that we shouldn't delay. Shall we create a protected getter and setter for `map` which redirects to the new property, and mark that deprecated?

Proposal: use longer names for the new properties that are more understandable for those creating custom subclasses: `languageIdMap`, `fileExtensionMap`

Reply:
WDYT about the naming proposal? And shall we make the properties `readonly`?

Reply:
Good idea, I've renamed the fields and made them readonly.